electrode-ota-server-dao-factory/src/factory.js

```diff
@@ -0,0 +1,322 @@
+import {alreadyExistsMsg} from 'electrode-ota-server-errors';
+import {promiseMap, reducer, toJSON} from 'electrode-ota-server-util';
+import _ from "lodash";
+
+const isEmpty = (arr) => {
+    if (arr == null) return true;
+    if (arr.length === 0) return true;
+    return false;
+};
+const isNotEmpty = (arr) => !isEmpty(arr);
+
+const apply = (target, source) => {
+    if (!source) {
+        return target;
+    }
+    for (const key of Object.keys(target)) {
+        if (key == '_validators') continue;
+        if (source.hasOwnProperty(key)) {
+            const newValue = source[key];
+            target[key] = newValue;
+        }
+    }
+    return target;
+};
+
+const ACCESSKEY = ["name", "id", "expires", "description", "lastAccess", "createdTime", "createdBy", "friendlyName"];
+
+const nullUnlessValue = (obj, keys) => {
+    for (const key of keys) {
+        if (obj[key] === void(0)) {
```

Comment:
you want to null properties that do not exist also?

don't like `undefined`?
 
I think `objDefaultNull` is slightly more telling.